test/io/test_io.py

```diff
@@ -998,3 +998,38 @@ def test_openapi_in_big_schema(defaultenv):
     with run(env=env) as postgrest:
         response = postgrest.session.get("/")
         assert response.status_code == 200
+
+
+@pytest.mark.parametrize("dburiType", ["noParams", "noParamsQMark", "withParams"])
+def test_get_pgrst_version_with_uri_connection_string(dburiType, dburi, defaultenv):
+    "The fallback_application_name should be added to the db-uri if it has a URI format"
+    defaultenv_without_libpq = {
+        key: value
+        for key, value in defaultenv.items()
+        if key not in ["PGDATABASE", "PGHOST", "PGUSER"]
+    }
+
+    env = {
+        "noParams": {**defaultenv, "PGRST_DB_URI": "postgresql://"},
+        "noParamsQMark": {**defaultenv, "PGRST_DB_URI": "postgresql://?"},
+        "withParams": {**defaultenv_without_libpq, "PGRST_DB_URI": dburi.decode()},
+    }
+
+    with run(env=env[dburiType]) as postgrest:
+        response = postgrest.session.post("/rpc/get_pgrst_version")
+        assert response.text.startswith('"PostgREST ')
+
+
+def test_get_pgrst_version_with_keyval_connection_string(defaultenv):
+    "The fallback_application_name should be added to the db-uri if it has a keyword/value format"
+    uri = f'dbname={defaultenv["PGDATABASE"]} host={defaultenv["PGHOST"]} user={defaultenv["PGUSER"]}'
+    defaultenv_without_libpq = {
+        key: value
+        for key, value in defaultenv.items()
+        if key not in ["PGDATABASE", "PGHOST", "PGUSER"]
+    }
+    env = {**defaultenv_without_libpq, "PGRST_DB_URI": uri}
+
+    with run(env=env) as postgrest:
+        response = postgrest.session.post("/rpc/get_pgrst_version")
+        assert response.text.startswith('"PostgREST ')
```

Comment:
Now that we escape the version, it'd be good to test the version is actually formatted in the way we expect.

I suggest putting the full version here and comparing it by obtaining the version from `postgrest.cabal`. The `commit` still will not be possible to compare without breaking later so that part can be left out.

Reply:
I saw that the `prettyVersion` is also used to return the version in the `Server` header. Maybe we could use that to compare (already added an example to the tests).